htsget/htsget.go

```diff
@@ -0,0 +1,238 @@
+package htsget
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/NBISweden/sda-cli/helpers"
+)
+
+// Help text and command line flags.
+
+// Usage text that will be displayed as command line help text when using the
+// `help htsget` command
+var Usage = `
+USAGE: %s htsget [-dataset <datasetID>] [-filename <filename>] (-reference <referenceName>) [-htsgethost <htsget-hostname>] [-key <public-key-file>] (-outdir <dir>)
+
+htsget:
+    Htsget downloads files from the Sensitive Data Archive (SDA), using the
+	htsget server. A dataset and a filename must be provided in order to 
+	download the file. The files will be downloaded in the current
+    directory, if outdir is not defined.
+`
+
+// ArgHelp is the suffix text that will be displayed after the argument list in
+// the module help
+var ArgHelp = `
+    [dataset]
+        The ID of the dataset that the file is part of.
+    [filename]
+        The name of the file to download.
+	[reference]
```

Comment:
why `[reference]`, `[htsgethost]` and `[key]` are more indented than the others?